### ISSUE TITLE
btrbk ssh cipher option

### ISSUE THREAD

**Issue by dloshakov** [digint/btrbk#47]
Hello, I've add some changes to btrbk, to be able to choose ssh cipher for better performance while send/recieve subvolume snapshots via ssh. I have added two ciphers to choose of aes128-ctr and arcfour256. Here are changes:
```
btrbk.conf:
ssh_cipher                      aes128-ctr

btrbk:
79a80
>    ssh_cipher        => { default => undef,    accept => [ "arcfour256", "aes128-ctr" ] },
238a240
>    my $ssh_cipher      = config_key($config, "ssh_cipher");
241a244
>   push(@ssh_options, '-c', $ssh_cipher) if ($ssh_cipher);
```
Speed improvements (without compression):
- default cipher ~28MiB/s
- aes128-ctr ~36MiB/s

Cheers!

**Comment by mac-linux-free** [comment]
cool...what about acrfour ?

**Comment by dloshakov** [comment]
My tests was done on Archlinux with btrfs-progs 4.1.2 and openssh 7.1, sad but in archlinux openssh doesn't have acrfour cipher.
```
/mnt/root_btrfs/vsol1
+++ /mnt/root_btrfs/_btrbk_snap/vsol1.20150905
!!! Target "{10.20.50.106}/mnt/root_btrfs/vsol1" aborted: Failed to fetch subvolume detail: Unable to negotiate with 10.20.50.106: no matching cipher found. Their offer: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
```

**Comment by digint** [comment]
There's so many options to ssh, and I don't think it makes much sense to copy them all into btrbk configuration, as they are all configurable system-wide in `/etc/ssh/ssh_config`.
Do you have any good reason why this is not enough?

**Comment by dloshakov** [comment]
Good question, but if I understand changes in /etc/ssh/ssh_config will change system-wide options, witch is not so good idea, to change cipher for whole system.
To be honest I don't know to put or not my changes to upstream code, I will let you decide :)

BTW thanks for excellent software!

**Comment by digint** [comment]
In ssh_config, you can change options on per-host basis:
```
Host myhost
   Cipher foo
```
On course this will still be a system-wide setting for this specific host.

I don't think it is a good idea to change such ssh settings outside ssh_config, as it is easily forgotten and if a cipher becomes weak, btrbk will still use it for years.

Still I see that there might be reasons to do this anyway, and supporting a `ssh_cipher` option in btrbk can be easily implemented, simply allowing just ANY cipher string.

**Comment by dloshakov** [comment]
Nice, thanks.

**Comment by digint** [comment]
added configuration option `ssh_cipher_spec`, allowing comma-separated list of ciphers (as does the ssh -c option).
Fixed in commit: dd6104339e548353bd2dc921eab8df35997fa3cb
